DataFormats/PortableVertex/interface/VertexSoA.h

```diff
@@ -0,0 +1,128 @@
+
+                      SOA_COLUMN(float, aux1),
+                      SOA_COLUMN(float, aux2),
+
+                      SOA_EIGEN_COLUMN(VertexToTrackInt, track_id),
+                      SOA_EIGEN_COLUMN(VertexToTrack, track_weight),
+
+                      SOA_COLUMN(bool, isGood),
+                      SOA_COLUMN(int, order),
+
+                      SOA_COLUMN(float, sw),
+                      SOA_COLUMN(float, se),
+                      SOA_COLUMN(float, swz),
+                      SOA_COLUMN(float, swE),
+                      SOA_COLUMN(float, exp),
+                      SOA_COLUMN(float, exparg),
+
+                      // Use entries for blocks
+                      SOA_COLUMN(int32_t, nV))
+
+  using VertexSoA = VertexSoALayout<>;
+
+  using TrackToVertex = Eigen::Vector<float, 512>;  // 512 is the max vertex allowed
+  GENERATE_SOA_LAYOUT(TrackSoALayout,
+                      // columns: one value per element
+                      SOA_COLUMN(float, dxy2),
+                      SOA_COLUMN(float, dxy2AtIP),
+                      SOA_COLUMN(float, dz2),
+                      SOA_COLUMN(float, oneoverdz2),
+                      SOA_COLUMN(float, weight),
+                      SOA_COLUMN(float, sum_Z),
+                      SOA_COLUMN(int, kmin),
+                      SOA_COLUMN(int, kmax),
+                      SOA_COLUMN(bool, isGood),
+                      SOA_COLUMN(int, order),
+                      SOA_COLUMN(int, tt_index),
+
+                      SOA_COLUMN(float, x),
+                      SOA_COLUMN(float, y),
+                      SOA_COLUMN(float, z),
+
+                      SOA_COLUMN(float, xAtIP),
+                      SOA_COLUMN(float, yAtIP),
+
+                      SOA_COLUMN(float, dx),
+                      SOA_COLUMN(float, dy),
+                      SOA_COLUMN(float, dz),
+
+                      SOA_COLUMN(float, dxError),
+                      SOA_COLUMN(float, dyError),
+                      SOA_COLUMN(float, dzError),
+
+                      SOA_COLUMN(float, px),
+                      SOA_COLUMN(float, py),
+                      SOA_COLUMN(float, pz),
+
+                      SOA_COLUMN(float, aux1),
+                      SOA_COLUMN(float, aux2),
+
+                      // Track-vertex association
+                      SOA_EIGEN_COLUMN(TrackToVertex, vert_sw),
+                      SOA_EIGEN_COLUMN(TrackToVertex, vert_se),
+                      SOA_EIGEN_COLUMN(TrackToVertex, vert_swz),
+                      SOA_EIGEN_COLUMN(TrackToVertex, vert_swE),
+                      SOA_EIGEN_COLUMN(TrackToVertex, vert_exp),
+                      SOA_EIGEN_COLUMN(TrackToVertex, vert_exparg),
+
+                      // scalars: one value for the whole structure
+                      SOA_SCALAR(int32_t, nT),
+                      SOA_SCALAR(float, totweight))
+
+  using TrackSoA = TrackSoALayout<>;
+
+  GENERATE_SOA_LAYOUT(ClusterParams,
```

Comment:
It looks like the `ClusterParams` is used only internally in `PrimaryVertexProducer_Alpaka`, and is not put into the Event. Then it would be better defined in `RecoVertex/PrimaryVertexProducer_Alpaka/plugins/alpaka`.